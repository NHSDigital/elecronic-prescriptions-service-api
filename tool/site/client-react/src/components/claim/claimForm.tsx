import {Button, Form} from "nhsuk-react-components"
import * as React from "react"
import {FieldArray, Formik} from "formik"
import {VALUE_SET_PRESCRIPTION_CHARGE_EXEMPTION} from "./reference-data/valueSets"
import Exemption from "./exemption"
import ProductArray from "./productArray"

export interface ClaimFormProps {
  products: Array<StaticProductInfo>
  sendClaim: (claim: ClaimFormValues) => void
}

const ClaimForm: React.FC<ClaimFormProps> = ({
  products,
  sendClaim
}) => {
  const initialValues: ClaimFormValues = {
    products: products.map(product => ({
      ...product,
      patientPaid: false,
      endorsements: []
    })),
    exemption: {
      code: VALUE_SET_PRESCRIPTION_CHARGE_EXEMPTION[0].code,
      evidenceSeen: false
    }
  }

  return (
    <Formik<ClaimFormValues> initialValues={initialValues} onSubmit={sendClaim}>
      {formik =>
        <Form onSubmit={formik.handleSubmit} onReset={formik.handleReset}>
          <FieldArray name="products" component={ProductArray}/>
          <Exemption name="exemption"/>
          <Button type="submit">Claim</Button>
          <Button type="reset" secondary>Reset</Button>
        </Form>
      }
    </Formik>
  )
}

export interface ClaimFormValues {
  products: Array<ProductFormValues>
  exemption: ExemptionFormValues
}

export interface StaticProductInfo {
  id: string
  name: string
  status: string
  quantityDispensed: string
}

export interface ProductFormValues extends StaticProductInfo {
  patientPaid: boolean
  endorsements: Array<EndorsementFormValues>
}

export interface EndorsementFormValues {
  code: string
  supportingInfo: string
}

export interface ExemptionFormValues {
  code: string
  evidenceSeen: boolean
}

export default ClaimForm
