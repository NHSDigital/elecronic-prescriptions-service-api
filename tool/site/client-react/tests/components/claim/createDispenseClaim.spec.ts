import {
  getMedicationDispenseResources,
  getMedicationRequestResources,
  getPatientResources
} from "../../../src/fhir/bundleResourceFinder"
import {createClaim} from "../../../src/components/claim/createDispenseClaim"
import * as fhir from "fhir/r4"
import {getClaimSequenceIdentifierExtension} from "../../../src/fhir/customExtensions"
import {readMessage} from "../../messages/messages"
import {createStaticProductInfoArray} from "../../../src/pages/claimPage"
import {ClaimFormValues} from "../../../src/components/claim/claimForm"
import {PRESCRIPTION_CHARGE_EXEMPTION_CODE_NONE} from "../../../src/fhir/reference-data/valueSets"

const prescriptionOrder = readMessage("prescriptionOrder.json")
const dispenseNotification = readMessage("dispenseNotification.json")
const patient = getPatientResources(prescriptionOrder)[0]
const medicationRequests = getMedicationRequestResources(prescriptionOrder)
const medicationDispenses = getMedicationDispenseResources(dispenseNotification)

test("Produces expected result when endorsement not present", () => {
  const staticProductInfoArray = createStaticProductInfoArray(medicationDispenses)
  const claimFormValues: ClaimFormValues = {
    products: staticProductInfoArray.map(staticProductInfo => ({
      ...staticProductInfo,
      patientPaid: true,
      endorsements: []
    })),
    exemption: {
      code: PRESCRIPTION_CHARGE_EXEMPTION_CODE_NONE,
      evidenceSeen: false
    }
  }
  const result = createClaim(patient, medicationRequests, medicationDispenses, claimFormValues)
  replaceNonDeterministicValues(result)
  expect(result).toMatchSnapshot()
})

test("Produces expected result when single endorsement present", () => {
  const staticProductInfoArray = createStaticProductInfoArray(medicationDispenses)
  const claimFormValues: ClaimFormValues = {
    products: staticProductInfoArray.map(staticProductInfo => ({
      ...staticProductInfo,
      patientPaid: true,
      endorsements: [{
        code: "IP",
        supportingInfo: "£210.91,100ml,Specials Ltd,Lic12345678,BN12345678"
      }]
    })),
    exemption: {
      code: PRESCRIPTION_CHARGE_EXEMPTION_CODE_NONE,
      evidenceSeen: false
    }
  }
  const result = createClaim(patient, medicationRequests, medicationDispenses, claimFormValues)
  replaceNonDeterministicValues(result)
  expect(result).toMatchSnapshot()
})

test("Produces expected result when multiple endorsements present", () => {
  const staticProductInfoArray = createStaticProductInfoArray(medicationDispenses)
  const claimFormValues: ClaimFormValues = {
    products: staticProductInfoArray.map(staticProductInfo => ({
      ...staticProductInfo,
      patientPaid: true,
      endorsements: [
        {
          code: "BB",
          supportingInfo: ""
        },
        {
          code: "IP",
          supportingInfo: "£210.91,100ml,Specials Ltd,Lic12345678,BN12345678"
        }
      ]
    })),
    exemption: {
      code: PRESCRIPTION_CHARGE_EXEMPTION_CODE_NONE,
      evidenceSeen: false
    }
  }
  const result = createClaim(patient, medicationRequests, medicationDispenses, claimFormValues)
  replaceNonDeterministicValues(result)
  expect(result).toMatchSnapshot()
})

test("Produces expected result when non-default exemption code selected", () => {
  const staticProductInfoArray = createStaticProductInfoArray(medicationDispenses)
  const claimFormValues: ClaimFormValues = {
    products: staticProductInfoArray.map(staticProductInfo => ({
      ...staticProductInfo,
      patientPaid: false,
      endorsements: []
    })),
    exemption: {
      code: "0005",
      evidenceSeen: true
    }
  }
  const result = createClaim(patient, medicationRequests, medicationDispenses, claimFormValues)
  replaceNonDeterministicValues(result)
  expect(result).toMatchSnapshot()
})

function replaceNonDeterministicValues(result: fhir.Claim) {
  result.created = "2021-11-01T14:35:00.000Z"
  result.identifier[0].value = "fb51061d-3808-4551-984d-4b25924cd15a"
  result.item
    .flatMap(item => item.detail)
    .map(detail => getClaimSequenceIdentifierExtension(detail.extension))
    .forEach(extension => extension.valueIdentifier.value = "fb51061d-3808-4551-984d-4b25924cd15a")
}
