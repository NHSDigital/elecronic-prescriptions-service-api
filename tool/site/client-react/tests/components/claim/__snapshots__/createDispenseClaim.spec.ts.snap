// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Produces expected result when endorsement not present 1`] = `
Object {
  "created": "2021-11-01T14:35:00.000Z",
  "identifier": Array [
    Object {
      "system": "https://tools.ietf.org/html/rfc4122",
      "value": "fb51061d-3808-4551-984d-4b25924cd15a",
    },
  ],
  "insurance": Array [
    Object {
      "coverage": Object {
        "display": "NHS BUSINESS SERVICES AUTHORITY",
        "identifier": Object {
          "system": "https://fhir.nhs.uk/Id/ods-organization-code",
          "value": "T1450",
        },
      },
      "focal": true,
      "sequence": 1,
    },
  ],
  "item": Array [
    Object {
      "detail": Array [
        Object {
          "extension": Array [
            Object {
              "url": "https://fhir.nhs.uk/StructureDefinition/Extension-ClaimSequenceIdentifier",
              "valueIdentifier": Object {
                "system": "https://fhir.nhs.uk/Id/claim-sequence-identifier",
                "value": "fb51061d-3808-4551-984d-4b25924cd15a",
              },
            },
            Object {
              "url": "https://fhir.nhs.uk/StructureDefinition/Extension-ClaimMedicationRequestReference",
              "valueReference": Object {
                "identifier": Object {
                  "system": "https://fhir.nhs.uk/Id/prescription-order-item-number",
                  "value": "a54219b8-f741-4c47-b662-e4f8dfa49ab6",
                },
              },
            },
            Object {
              "extension": Array [
                Object {
                  "url": "numberOfRepeatsIssued",
                  "valueInteger": 1,
                },
                Object {
                  "url": "numberOfRepeatsAllowed",
                  "valueInteger": 6,
                },
              ],
              "url": "https://fhir.nhs.uk/StructureDefinition/Extension-EPS-RepeatInformation",
            },
          ],
          "modifier": Array [
            Object {
              "coding": Array [
                Object {
                  "code": "0001",
                  "display": "Item fully dispensed",
                  "system": "https://fhir.nhs.uk/CodeSystem/medicationdispense-type",
                },
              ],
            },
          ],
          "productOrService": Object {
            "coding": Array [
              Object {
                "code": "15517911000001104",
                "display": "Methotrexate 10mg/0.2ml solution for injection pre-filled syringes",
                "system": "http://snomed.info/sct",
              },
            ],
          },
          "quantity": Object {
            "code": "428673006",
            "system": "http://snomed.info/sct",
            "unit": "tablet",
            "value": 63,
          },
          "sequence": 1,
          "subDetail": Array [
            Object {
              "productOrService": Object {
                "coding": Array [
                  Object {
                    "code": "15517911000001104",
                    "display": "Methotrexate 10mg/0.2ml solution for injection pre-filled syringes",
                    "system": "http://snomed.info/sct",
                  },
                ],
              },
              "programCode": Array [
                Object {
                  "coding": Array [
                    Object {
                      "code": "NDEC",
                      "display": "No Dispenser Endorsement Code",
                      "system": "https://fhir.nhs.uk/CodeSystem/medicationdispense-endorsement",
                    },
                  ],
                },
                Object {
                  "coding": Array [
                    Object {
                      "code": "paid-once",
                      "display": "Paid Once",
                      "system": "https://fhir.nhs.uk/CodeSystem/DM-prescription-charge",
                    },
                  ],
                },
              ],
              "quantity": Object {
                "code": "428673006",
                "system": "http://snomed.info/sct",
                "unit": "tablet",
                "value": 63,
              },
              "sequence": 1,
            },
          ],
        },
      ],
      "extension": Array [
        Object {
          "url": "https://fhir.nhs.uk/StructureDefinition/Extension-EPS-TaskBusinessStatus",
          "valueCoding": Object {
            "code": "0006",
            "display": "Dispensed",
            "system": "https://fhir.nhs.uk/CodeSystem/EPS-task-business-status",
          },
        },
      ],
      "productOrService": Object {
        "coding": Array [
          Object {
            "code": "16076005",
            "display": "Prescription",
            "system": "http://snomed.info/sct",
          },
        ],
      },
      "programCode": Array [
        Object {
          "coding": Array [
            Object {
              "code": "0001",
              "display": "Patient has paid appropriate charges",
              "system": "https://fhir.nhs.uk/CodeSystem/prescription-charge-exemption",
            },
          ],
        },
        Object {
          "coding": Array [
            Object {
              "code": "no-evidence-seen",
              "display": "No Evidence Seen",
              "system": "https://fhir.nhs.uk/CodeSystem/DM-exemption-evidence",
            },
          ],
        },
      ],
      "sequence": 1,
    },
  ],
  "patient": Object {
    "identifier": Object {
      "system": "https://fhir.nhs.uk/Id/nhs-number",
      "value": "9449305552",
    },
  },
  "payee": Object {
    "party": Object {
      "display": "HEATH PHARMACY",
      "identifier": Object {
        "system": "https://fhir.nhs.uk/Id/ods-organization-code",
        "value": "FER21",
      },
      "type": "Organization",
    },
    "type": Object {
      "coding": Array [
        Object {
          "code": "provider",
          "display": "Provider",
          "system": "http://terminology.hl7.org/CodeSystem/payeetype",
        },
      ],
    },
  },
  "prescription": Object {
    "extension": Array [
      Object {
        "extension": Array [
          Object {
            "url": "shortForm",
            "valueIdentifier": Object {
              "system": "https://fhir.nhs.uk/Id/prescription-order-number",
              "value": "B407F7-A99968-4DB2AU",
            },
          },
          Object {
            "url": "UUID",
            "valueIdentifier": Object {
              "system": "https://fhir.nhs.uk/Id/prescription",
              "value": "10d37820-dda2-4810-94d7-c79e5fdb4b46",
            },
          },
        ],
        "url": "https://fhir.nhs.uk/StructureDefinition/Extension-DM-GroupIdentifier",
      },
    ],
  },
  "priority": Object {
    "coding": Array [
      Object {
        "code": "normal",
        "system": "http://terminology.hl7.org/CodeSystem/processpriority",
      },
    ],
  },
  "provider": Object {
    "display": "Mr Peter Potion",
    "identifier": Object {
      "system": "https://fhir.hl7.org.uk/Id/gphc-number",
      "value": "7654321",
    },
    "type": "Practitioner",
  },
  "resourceType": "Claim",
  "status": "active",
  "type": Object {
    "coding": Array [
      Object {
        "code": "pharmacy",
        "display": "Pharmacy",
        "system": "http://terminology.hl7.org/CodeSystem/claim-type",
      },
    ],
  },
  "use": "claim",
}
`;

exports[`Produces expected result when multiple endorsements present 1`] = `
Object {
  "created": "2021-11-01T14:35:00.000Z",
  "identifier": Array [
    Object {
      "system": "https://tools.ietf.org/html/rfc4122",
      "value": "fb51061d-3808-4551-984d-4b25924cd15a",
    },
  ],
  "insurance": Array [
    Object {
      "coverage": Object {
        "display": "NHS BUSINESS SERVICES AUTHORITY",
        "identifier": Object {
          "system": "https://fhir.nhs.uk/Id/ods-organization-code",
          "value": "T1450",
        },
      },
      "focal": true,
      "sequence": 1,
    },
  ],
  "item": Array [
    Object {
      "detail": Array [
        Object {
          "extension": Array [
            Object {
              "url": "https://fhir.nhs.uk/StructureDefinition/Extension-ClaimSequenceIdentifier",
              "valueIdentifier": Object {
                "system": "https://fhir.nhs.uk/Id/claim-sequence-identifier",
                "value": "fb51061d-3808-4551-984d-4b25924cd15a",
              },
            },
            Object {
              "url": "https://fhir.nhs.uk/StructureDefinition/Extension-ClaimMedicationRequestReference",
              "valueReference": Object {
                "identifier": Object {
                  "system": "https://fhir.nhs.uk/Id/prescription-order-item-number",
                  "value": "a54219b8-f741-4c47-b662-e4f8dfa49ab6",
                },
              },
            },
            Object {
              "extension": Array [
                Object {
                  "url": "numberOfRepeatsIssued",
                  "valueInteger": 1,
                },
                Object {
                  "url": "numberOfRepeatsAllowed",
                  "valueInteger": 6,
                },
              ],
              "url": "https://fhir.nhs.uk/StructureDefinition/Extension-EPS-RepeatInformation",
            },
          ],
          "modifier": Array [
            Object {
              "coding": Array [
                Object {
                  "code": "0001",
                  "display": "Item fully dispensed",
                  "system": "https://fhir.nhs.uk/CodeSystem/medicationdispense-type",
                },
              ],
            },
          ],
          "productOrService": Object {
            "coding": Array [
              Object {
                "code": "15517911000001104",
                "display": "Methotrexate 10mg/0.2ml solution for injection pre-filled syringes",
                "system": "http://snomed.info/sct",
              },
            ],
          },
          "quantity": Object {
            "code": "428673006",
            "system": "http://snomed.info/sct",
            "unit": "tablet",
            "value": 63,
          },
          "sequence": 1,
          "subDetail": Array [
            Object {
              "productOrService": Object {
                "coding": Array [
                  Object {
                    "code": "15517911000001104",
                    "display": "Methotrexate 10mg/0.2ml solution for injection pre-filled syringes",
                    "system": "http://snomed.info/sct",
                  },
                ],
              },
              "programCode": Array [
                Object {
                  "coding": Array [
                    Object {
                      "code": "BB",
                      "display": "Broken Bulk",
                      "system": "https://fhir.nhs.uk/CodeSystem/medicationdispense-endorsement",
                    },
                  ],
                },
                Object {
                  "coding": Array [
                    Object {
                      "code": "IP",
                      "display": "Invoice Price for less common products or special items",
                      "system": "https://fhir.nhs.uk/CodeSystem/medicationdispense-endorsement",
                    },
                  ],
                  "text": "£210.91,100ml,Specials Ltd,Lic12345678,BN12345678",
                },
                Object {
                  "coding": Array [
                    Object {
                      "code": "paid-once",
                      "display": "Paid Once",
                      "system": "https://fhir.nhs.uk/CodeSystem/DM-prescription-charge",
                    },
                  ],
                },
              ],
              "quantity": Object {
                "code": "428673006",
                "system": "http://snomed.info/sct",
                "unit": "tablet",
                "value": 63,
              },
              "sequence": 1,
            },
          ],
        },
      ],
      "extension": Array [
        Object {
          "url": "https://fhir.nhs.uk/StructureDefinition/Extension-EPS-TaskBusinessStatus",
          "valueCoding": Object {
            "code": "0006",
            "display": "Dispensed",
            "system": "https://fhir.nhs.uk/CodeSystem/EPS-task-business-status",
          },
        },
      ],
      "productOrService": Object {
        "coding": Array [
          Object {
            "code": "16076005",
            "display": "Prescription",
            "system": "http://snomed.info/sct",
          },
        ],
      },
      "programCode": Array [
        Object {
          "coding": Array [
            Object {
              "code": "0001",
              "display": "Patient has paid appropriate charges",
              "system": "https://fhir.nhs.uk/CodeSystem/prescription-charge-exemption",
            },
          ],
        },
        Object {
          "coding": Array [
            Object {
              "code": "no-evidence-seen",
              "display": "No Evidence Seen",
              "system": "https://fhir.nhs.uk/CodeSystem/DM-exemption-evidence",
            },
          ],
        },
      ],
      "sequence": 1,
    },
  ],
  "patient": Object {
    "identifier": Object {
      "system": "https://fhir.nhs.uk/Id/nhs-number",
      "value": "9449305552",
    },
  },
  "payee": Object {
    "party": Object {
      "display": "HEATH PHARMACY",
      "identifier": Object {
        "system": "https://fhir.nhs.uk/Id/ods-organization-code",
        "value": "FER21",
      },
      "type": "Organization",
    },
    "type": Object {
      "coding": Array [
        Object {
          "code": "provider",
          "display": "Provider",
          "system": "http://terminology.hl7.org/CodeSystem/payeetype",
        },
      ],
    },
  },
  "prescription": Object {
    "extension": Array [
      Object {
        "extension": Array [
          Object {
            "url": "shortForm",
            "valueIdentifier": Object {
              "system": "https://fhir.nhs.uk/Id/prescription-order-number",
              "value": "B407F7-A99968-4DB2AU",
            },
          },
          Object {
            "url": "UUID",
            "valueIdentifier": Object {
              "system": "https://fhir.nhs.uk/Id/prescription",
              "value": "10d37820-dda2-4810-94d7-c79e5fdb4b46",
            },
          },
        ],
        "url": "https://fhir.nhs.uk/StructureDefinition/Extension-DM-GroupIdentifier",
      },
    ],
  },
  "priority": Object {
    "coding": Array [
      Object {
        "code": "normal",
        "system": "http://terminology.hl7.org/CodeSystem/processpriority",
      },
    ],
  },
  "provider": Object {
    "display": "Mr Peter Potion",
    "identifier": Object {
      "system": "https://fhir.hl7.org.uk/Id/gphc-number",
      "value": "7654321",
    },
    "type": "Practitioner",
  },
  "resourceType": "Claim",
  "status": "active",
  "type": Object {
    "coding": Array [
      Object {
        "code": "pharmacy",
        "display": "Pharmacy",
        "system": "http://terminology.hl7.org/CodeSystem/claim-type",
      },
    ],
  },
  "use": "claim",
}
`;

exports[`Produces expected result when non-default exemption code selected 1`] = `
Object {
  "created": "2021-11-01T14:35:00.000Z",
  "identifier": Array [
    Object {
      "system": "https://tools.ietf.org/html/rfc4122",
      "value": "fb51061d-3808-4551-984d-4b25924cd15a",
    },
  ],
  "insurance": Array [
    Object {
      "coverage": Object {
        "display": "NHS BUSINESS SERVICES AUTHORITY",
        "identifier": Object {
          "system": "https://fhir.nhs.uk/Id/ods-organization-code",
          "value": "T1450",
        },
      },
      "focal": true,
      "sequence": 1,
    },
  ],
  "item": Array [
    Object {
      "detail": Array [
        Object {
          "extension": Array [
            Object {
              "url": "https://fhir.nhs.uk/StructureDefinition/Extension-ClaimSequenceIdentifier",
              "valueIdentifier": Object {
                "system": "https://fhir.nhs.uk/Id/claim-sequence-identifier",
                "value": "fb51061d-3808-4551-984d-4b25924cd15a",
              },
            },
            Object {
              "url": "https://fhir.nhs.uk/StructureDefinition/Extension-ClaimMedicationRequestReference",
              "valueReference": Object {
                "identifier": Object {
                  "system": "https://fhir.nhs.uk/Id/prescription-order-item-number",
                  "value": "a54219b8-f741-4c47-b662-e4f8dfa49ab6",
                },
              },
            },
            Object {
              "extension": Array [
                Object {
                  "url": "numberOfRepeatsIssued",
                  "valueInteger": 1,
                },
                Object {
                  "url": "numberOfRepeatsAllowed",
                  "valueInteger": 6,
                },
              ],
              "url": "https://fhir.nhs.uk/StructureDefinition/Extension-EPS-RepeatInformation",
            },
          ],
          "modifier": Array [
            Object {
              "coding": Array [
                Object {
                  "code": "0001",
                  "display": "Item fully dispensed",
                  "system": "https://fhir.nhs.uk/CodeSystem/medicationdispense-type",
                },
              ],
            },
          ],
          "productOrService": Object {
            "coding": Array [
              Object {
                "code": "15517911000001104",
                "display": "Methotrexate 10mg/0.2ml solution for injection pre-filled syringes",
                "system": "http://snomed.info/sct",
              },
            ],
          },
          "quantity": Object {
            "code": "428673006",
            "system": "http://snomed.info/sct",
            "unit": "tablet",
            "value": 63,
          },
          "sequence": 1,
          "subDetail": Array [
            Object {
              "productOrService": Object {
                "coding": Array [
                  Object {
                    "code": "15517911000001104",
                    "display": "Methotrexate 10mg/0.2ml solution for injection pre-filled syringes",
                    "system": "http://snomed.info/sct",
                  },
                ],
              },
              "programCode": Array [
                Object {
                  "coding": Array [
                    Object {
                      "code": "NDEC",
                      "display": "No Dispenser Endorsement Code",
                      "system": "https://fhir.nhs.uk/CodeSystem/medicationdispense-endorsement",
                    },
                  ],
                },
                Object {
                  "coding": Array [
                    Object {
                      "code": "not-paid",
                      "display": "Not Paid",
                      "system": "https://fhir.nhs.uk/CodeSystem/DM-prescription-charge",
                    },
                  ],
                },
              ],
              "quantity": Object {
                "code": "428673006",
                "system": "http://snomed.info/sct",
                "unit": "tablet",
                "value": 63,
              },
              "sequence": 1,
            },
          ],
        },
      ],
      "extension": Array [
        Object {
          "url": "https://fhir.nhs.uk/StructureDefinition/Extension-EPS-TaskBusinessStatus",
          "valueCoding": Object {
            "code": "0006",
            "display": "Dispensed",
            "system": "https://fhir.nhs.uk/CodeSystem/EPS-task-business-status",
          },
        },
      ],
      "productOrService": Object {
        "coding": Array [
          Object {
            "code": "16076005",
            "display": "Prescription",
            "system": "http://snomed.info/sct",
          },
        ],
      },
      "programCode": Array [
        Object {
          "coding": Array [
            Object {
              "code": "0005",
              "display": "has a valid maternity exemption certificate",
              "system": "https://fhir.nhs.uk/CodeSystem/prescription-charge-exemption",
            },
          ],
        },
        Object {
          "coding": Array [
            Object {
              "code": "evidence-seen",
              "display": "Evidence Seen",
              "system": "https://fhir.nhs.uk/CodeSystem/DM-exemption-evidence",
            },
          ],
        },
      ],
      "sequence": 1,
    },
  ],
  "patient": Object {
    "identifier": Object {
      "system": "https://fhir.nhs.uk/Id/nhs-number",
      "value": "9449305552",
    },
  },
  "payee": Object {
    "party": Object {
      "display": "HEATH PHARMACY",
      "identifier": Object {
        "system": "https://fhir.nhs.uk/Id/ods-organization-code",
        "value": "FER21",
      },
      "type": "Organization",
    },
    "type": Object {
      "coding": Array [
        Object {
          "code": "provider",
          "display": "Provider",
          "system": "http://terminology.hl7.org/CodeSystem/payeetype",
        },
      ],
    },
  },
  "prescription": Object {
    "extension": Array [
      Object {
        "extension": Array [
          Object {
            "url": "shortForm",
            "valueIdentifier": Object {
              "system": "https://fhir.nhs.uk/Id/prescription-order-number",
              "value": "B407F7-A99968-4DB2AU",
            },
          },
          Object {
            "url": "UUID",
            "valueIdentifier": Object {
              "system": "https://fhir.nhs.uk/Id/prescription",
              "value": "10d37820-dda2-4810-94d7-c79e5fdb4b46",
            },
          },
        ],
        "url": "https://fhir.nhs.uk/StructureDefinition/Extension-DM-GroupIdentifier",
      },
    ],
  },
  "priority": Object {
    "coding": Array [
      Object {
        "code": "normal",
        "system": "http://terminology.hl7.org/CodeSystem/processpriority",
      },
    ],
  },
  "provider": Object {
    "display": "Mr Peter Potion",
    "identifier": Object {
      "system": "https://fhir.hl7.org.uk/Id/gphc-number",
      "value": "7654321",
    },
    "type": "Practitioner",
  },
  "resourceType": "Claim",
  "status": "active",
  "type": Object {
    "coding": Array [
      Object {
        "code": "pharmacy",
        "display": "Pharmacy",
        "system": "http://terminology.hl7.org/CodeSystem/claim-type",
      },
    ],
  },
  "use": "claim",
}
`;

exports[`Produces expected result when single endorsement present 1`] = `
Object {
  "created": "2021-11-01T14:35:00.000Z",
  "identifier": Array [
    Object {
      "system": "https://tools.ietf.org/html/rfc4122",
      "value": "fb51061d-3808-4551-984d-4b25924cd15a",
    },
  ],
  "insurance": Array [
    Object {
      "coverage": Object {
        "display": "NHS BUSINESS SERVICES AUTHORITY",
        "identifier": Object {
          "system": "https://fhir.nhs.uk/Id/ods-organization-code",
          "value": "T1450",
        },
      },
      "focal": true,
      "sequence": 1,
    },
  ],
  "item": Array [
    Object {
      "detail": Array [
        Object {
          "extension": Array [
            Object {
              "url": "https://fhir.nhs.uk/StructureDefinition/Extension-ClaimSequenceIdentifier",
              "valueIdentifier": Object {
                "system": "https://fhir.nhs.uk/Id/claim-sequence-identifier",
                "value": "fb51061d-3808-4551-984d-4b25924cd15a",
              },
            },
            Object {
              "url": "https://fhir.nhs.uk/StructureDefinition/Extension-ClaimMedicationRequestReference",
              "valueReference": Object {
                "identifier": Object {
                  "system": "https://fhir.nhs.uk/Id/prescription-order-item-number",
                  "value": "a54219b8-f741-4c47-b662-e4f8dfa49ab6",
                },
              },
            },
            Object {
              "extension": Array [
                Object {
                  "url": "numberOfRepeatsIssued",
                  "valueInteger": 1,
                },
                Object {
                  "url": "numberOfRepeatsAllowed",
                  "valueInteger": 6,
                },
              ],
              "url": "https://fhir.nhs.uk/StructureDefinition/Extension-EPS-RepeatInformation",
            },
          ],
          "modifier": Array [
            Object {
              "coding": Array [
                Object {
                  "code": "0001",
                  "display": "Item fully dispensed",
                  "system": "https://fhir.nhs.uk/CodeSystem/medicationdispense-type",
                },
              ],
            },
          ],
          "productOrService": Object {
            "coding": Array [
              Object {
                "code": "15517911000001104",
                "display": "Methotrexate 10mg/0.2ml solution for injection pre-filled syringes",
                "system": "http://snomed.info/sct",
              },
            ],
          },
          "quantity": Object {
            "code": "428673006",
            "system": "http://snomed.info/sct",
            "unit": "tablet",
            "value": 63,
          },
          "sequence": 1,
          "subDetail": Array [
            Object {
              "productOrService": Object {
                "coding": Array [
                  Object {
                    "code": "15517911000001104",
                    "display": "Methotrexate 10mg/0.2ml solution for injection pre-filled syringes",
                    "system": "http://snomed.info/sct",
                  },
                ],
              },
              "programCode": Array [
                Object {
                  "coding": Array [
                    Object {
                      "code": "IP",
                      "display": "Invoice Price for less common products or special items",
                      "system": "https://fhir.nhs.uk/CodeSystem/medicationdispense-endorsement",
                    },
                  ],
                  "text": "£210.91,100ml,Specials Ltd,Lic12345678,BN12345678",
                },
                Object {
                  "coding": Array [
                    Object {
                      "code": "paid-once",
                      "display": "Paid Once",
                      "system": "https://fhir.nhs.uk/CodeSystem/DM-prescription-charge",
                    },
                  ],
                },
              ],
              "quantity": Object {
                "code": "428673006",
                "system": "http://snomed.info/sct",
                "unit": "tablet",
                "value": 63,
              },
              "sequence": 1,
            },
          ],
        },
      ],
      "extension": Array [
        Object {
          "url": "https://fhir.nhs.uk/StructureDefinition/Extension-EPS-TaskBusinessStatus",
          "valueCoding": Object {
            "code": "0006",
            "display": "Dispensed",
            "system": "https://fhir.nhs.uk/CodeSystem/EPS-task-business-status",
          },
        },
      ],
      "productOrService": Object {
        "coding": Array [
          Object {
            "code": "16076005",
            "display": "Prescription",
            "system": "http://snomed.info/sct",
          },
        ],
      },
      "programCode": Array [
        Object {
          "coding": Array [
            Object {
              "code": "0001",
              "display": "Patient has paid appropriate charges",
              "system": "https://fhir.nhs.uk/CodeSystem/prescription-charge-exemption",
            },
          ],
        },
        Object {
          "coding": Array [
            Object {
              "code": "no-evidence-seen",
              "display": "No Evidence Seen",
              "system": "https://fhir.nhs.uk/CodeSystem/DM-exemption-evidence",
            },
          ],
        },
      ],
      "sequence": 1,
    },
  ],
  "patient": Object {
    "identifier": Object {
      "system": "https://fhir.nhs.uk/Id/nhs-number",
      "value": "9449305552",
    },
  },
  "payee": Object {
    "party": Object {
      "display": "HEATH PHARMACY",
      "identifier": Object {
        "system": "https://fhir.nhs.uk/Id/ods-organization-code",
        "value": "FER21",
      },
      "type": "Organization",
    },
    "type": Object {
      "coding": Array [
        Object {
          "code": "provider",
          "display": "Provider",
          "system": "http://terminology.hl7.org/CodeSystem/payeetype",
        },
      ],
    },
  },
  "prescription": Object {
    "extension": Array [
      Object {
        "extension": Array [
          Object {
            "url": "shortForm",
            "valueIdentifier": Object {
              "system": "https://fhir.nhs.uk/Id/prescription-order-number",
              "value": "B407F7-A99968-4DB2AU",
            },
          },
          Object {
            "url": "UUID",
            "valueIdentifier": Object {
              "system": "https://fhir.nhs.uk/Id/prescription",
              "value": "10d37820-dda2-4810-94d7-c79e5fdb4b46",
            },
          },
        ],
        "url": "https://fhir.nhs.uk/StructureDefinition/Extension-DM-GroupIdentifier",
      },
    ],
  },
  "priority": Object {
    "coding": Array [
      Object {
        "code": "normal",
        "system": "http://terminology.hl7.org/CodeSystem/processpriority",
      },
    ],
  },
  "provider": Object {
    "display": "Mr Peter Potion",
    "identifier": Object {
      "system": "https://fhir.hl7.org.uk/Id/gphc-number",
      "value": "7654321",
    },
    "type": "Practitioner",
  },
  "resourceType": "Claim",
  "status": "active",
  "type": Object {
    "coding": Array [
      Object {
        "code": "pharmacy",
        "display": "Pharmacy",
        "system": "http://terminology.hl7.org/CodeSystem/claim-type",
      },
    ],
  },
  "use": "claim",
}
`;
